### ISSUE TITLE
Requirements are wrong

### ISSUE THREAD

**Issue by oarangob** [taresh18/TTSizer#2]
The requirements.txt file includes items which are files hosted in the author´s computer and therefore, an immediate error is triggered.

**Comment by ConvoCortex** [comment]
I warn you that I can't tell if it's optimal for performance or even secure. It's based on what I learned when installing realtimestt and f5tts github repos and what LLM told me. took me some 6 hours to get it working.



I found a way to install this on windows, python 3.10.9, nvidia gpu.
this is more or less how. i mean i hope maybe you can just grab my freezed requirements.txt to get it working. i didnt reinstall it fresh to check if its exact instructions but maybe the  requirements.txt  will do job for you. but i guess LLM can help you finish it to make it work for you given all this info:




E:\programs\Python\Python3109\python.exe -m venv E:\programs\TTSizer\venv
cd E:\programs\TTSizer
venv\Scripts\activate
python.exe -m pip install --upgrade pip

make sure you have visual studio community 2022 with c++ desktop development kit
type in cmd before pip commands:
"C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\Build\vcvars64.bat"

pip cache purge
pip install torch==2.5.1+cu121 torchaudio==2.5.1+cu121 torchvision==0.20.1+cu121 --index-url https://download.pytorch.org/whl/cu121
pip install nemo-toolkit[asr] google-generativeai faster-whisper transformers datasets librosa soundfile audioread pydub ffmpeg-python speechbrain
pip install git+https://github.com/MahmoudAshraf97/ctc-forced-aligner.git git+https://github.com/oliverguhr/deepmultilingualpunctuation.git git+https://github.com/wenet-e2e/wespeaker.git

Open E:\programs\TTSizer\venv\lib\site-packages\nemo\utils\exp_manager.py, go to line 176, replace SIGKILL with SIGTERM, save.

https://huggingface.co/pcunwa/Kim-Mel-Band-Roformer-FT/tree/e42d447e5026e85492945b73438b7a98b9314347
Download kimmel_unwa_ft2_bleedless.ckpt - that's the one the code is looking for.
So download kimmel_unwa_ft2_bleedless.ckpt (913 MB), put it in E:\programs\TTSizer\TTSizer\weights\


The wespeaker API changed. load_model_local is the old function name.

Fix in E:\programs\TTSizer\TTSizer\ttsizer\core\outlier_detect.py line 57:

Change:

python
model = wespeaker.load_model_local(str(self.model_path))

To:

python
model = wespeaker.load_model('english')  # or use load_model_pt for local pytorch mo

python -c "import torch; print(torch.cuda.is_available())"

and configure, get gemini api key from google ai studio 

update in config to 2.5 flash
# 4. LLM Diarization & Initial Transcription
llm_diarizer:
  output_folder: "transcriptions"
  model_name: "gemini-2.5-flash"  # LLM model for diarization

and run

(venv) E:\programs\TTSizer\TTSizer>python -m ttsizer.main

I don't know if it's necessary for this repo but you might have to setup cuda and cudnn. here is a note i made to myself about it once:
install cuda 12.1 https://developer.nvidia.com/cuda-12-1-0-download-archive?target_os=Windows&target_arch=x86_64&target_version=11&target_type=exe_local 
custom installation without driver or anything else. only cuda. 
add to path: E:\programs\CUDA 12.1\bin  and   E:\programs\CUDA 12.1\libnvvp (I think in libvvp you put the cudnn files? don't remember)

notepad ttsizer\main.py
Go to line 190 and uncomment this line:
python
diarizer.process_directory(input_path, output_path)



one issue with repo : in order to start interrupted workflow have to remove all files from output/audio or else it will hang on step 1. could add some skip feature for that step. the llm dializer via gemini im currently testing with 3 batch api as the realtime wont process files 4+ minutes reliably due to api timeouts. but im currently rate limited.  i might fork it later with my changes to skipping and gemini and added yt-dlp and maybe even f5tts training idk

and grab my reworked requirements.txt from my freeze


absl-py==2.2.2
accelerate==1.10.1
aiohappyeyeballs==2.6.1
aiohttp==3.11.18
aiosignal==1.3.2
alembic==1.15.2
annotated-types==0.7.0
antlr4-python3-runtime==4.9.3
anyio==4.9.0
asteroid==0.7.0
asteroid-filterbanks==0.4.0
asttokens==3.0.0
async-timeout==5.0.1
attrs==25.3.0
audiomentations==0.24.0
audioread==3.0.1
auraloss==0.4.0
av==15.1.0
backports-datetime-fromisoformat==2.0.3
beartype==0.14.1
bitsandbytes==0.45.5
braceexpand==0.1.7
Brotli==1.1.0
cached-property==2.0.1
cachetools==5.5.2
certifi==2024.2.2
cffi==1.17.1
charset-normalizer==3.4.0
click==8.1.8
cloudpickle==3.1.1
colorama==0.4.6
coloredlogs==15.0.1
colorlog==6.9.0
comm==0.2.2
contourpy==1.3.2
ctc-forced-aligner @ git+https://github.com/MahmoudAshraf97/ctc-forced-aligner.git@1f0a5f860d3d9daf3d94edb1c7d18f90d1702e5b
ctranslate2==4.6.0
cuda-bindings==13.0.1
cuda-pathfinder==1.3.0
cuda-python==13.0.1
cycler==0.12.1
Cython==0.29.37
cytoolz==1.0.1
dataclassy==0.11.1
datasets==3.2.0
decorator==5.1.1
deepmultilingualpunctuation @ git+https://github.com/oliverguhr/deepmultilingualpunctuation.git@5a0dd7f4fd56687f59405aa8eba1144393d8b74b
demucs==4.0.0
diffq==0.2.4
dill==0.3.8
Distance==0.1.3
docker-pycreds==0.4.0
docopt==0.6.2
dora_search==0.1.12
editdistance==0.8.1
efficientnet_pytorch==0.7.1
einops==0.8.1
et-xmlfile==1.1.0
exceptiongroup==1.2.2
executing==2.2.0
faster-whisper==1.1.1
ffmpeg-python==0.2.0
fiddle==0.3.0
filelock==3.18.0
flatbuffers==25.2.10
fonttools==4.60.1
frozenlist==1.6.0
fsspec==2024.9.0
future==1.0.0
g2p-en==2.1.0
gitdb==4.0.12
GitPython==3.1.44
google-ai-generativelanguage==0.6.15
google-api-core==2.25.0rc1
google-api-python-client==2.169.0
google-auth==2.39.0
google-auth-httplib2==0.2.0
google-genai==1.41.0
google-generativeai==0.8.5
googleapis-common-protos==1.70.0
graphviz==0.20.3
greenlet==3.1.1
grpcio==1.71.0
grpcio-status==1.71.0
h11==0.16.0
hdbscan==0.8.37
httpcore==1.0.9
httplib2==0.22.0
httpx==0.28.1
huggingface-hub==0.35.3
humanfriendly==10.0
hydra-core==1.3.2
hyper-connections==0.1.11
HyperPyYAML==1.2.2
idna==3.10
indic_numtowords==1.1.0
inflect==7.5.0
intervaltree==3.1.0
ipython==8.36.0
ipywidgets==8.1.6
jedi==0.19.2
Jinja2==3.1.6
jiwer==3.1.0
joblib==1.4.2
julius==0.2.7
jupyterlab_widgets==3.0.14
kaldi-python-io==1.2.2
kaldiio==2.18.1
keyboard==0.13.5
kiwisolver==1.4.9
kneed==0.8.5
lameenc==1.8.1
lazy_loader==0.4
Levenshtein==0.27.1
lhotse==1.31.1
libcst==1.7.0
librosa==0.9.2
lightning==2.4.0
lightning-utilities==0.14.3
lilcom==1.8.1
llvmlite==0.44.0
loguru==0.7.3
loralib==0.1.2
Mako==1.3.10
Markdown==3.8
markdown-it-py==3.0.0
MarkupSafe==3.0.2
marshmallow==4.0.0
matplotlib==3.10.0
matplotlib-inline==0.1.7
mdurl==0.1.2
mediapy==1.1.6
mir_eval==0.8.2
ml_collections==1.1.0
ml_dtypes==0.5.3
more-itertools==10.7.0
mpmath==1.3.0
msgpack==1.1.0
multidict==6.4.3
multiprocess==0.70.16
munch==4.0.0
nemo-toolkit==2.2.1
networkx==3.4.2
nltk==3.9.1
num2words==0.5.14
numba==0.61.0
numpy==1.26.4
omegaconf==2.2.3
onnx==1.17.0
onnxruntime==1.23.0
openpyxl==3.1.2
openunmix==1.3.0
optuna==4.5.0
packaging==24.2
pandas==2.3.3
parso==0.8.4
pb-bss-eval==0.0.2
pedalboard==0.8.9
peft==0.17.1
pesq==0.0.4
pillow==11.0.0
plac==1.4.5
platformdirs==4.3.6
pooch==1.8.2
praatio==6.2.0
pretrainedmodels==0.7.4
primePy==1.3
prodigyopt==1.1.2
prompt_toolkit==3.0.51
propcache==0.3.1
proto-plus==1.26.1
protobuf==3.20.3
psutil==7.0.0
pure_eval==0.2.3
pyannote.core==5.0.0
pyannote.database==5.1.3
pyannote.metrics==3.2.1
pyarrow==20.0.0
pyasn1==0.6.1
pyasn1_modules==0.4.2
pybind11==2.13.6
pycparser==2.22
pydantic==2.11.3
pydantic_core==2.33.1
pydub==0.25.1
Pygments==2.18.0
pyloudnorm==0.1.1
pynndescent==0.5.13
pyparsing==3.2.1
pyreadline3==3.5.4
PySocks==1.7.1
pystoi==0.4.1
python-dateutil==2.9.0
python-dotenv==1.1.0
pytorch-lightning==2.5.5
pytorch-ranger==0.1.1
pytz==2025.2
pywin32==311
PyYAML==6.0.2
RapidFuzz==3.13.0
regex==2024.11.6
requests==2.32.3
resampy==0.4.3
retrying==1.3.4
rich==13.9.4
rich-click==1.8.5
rotary-embedding-torch==0.3.5
rsa==4.9.1
ruamel.yaml==0.18.10
ruamel.yaml.clib==0.2.12
sacremoses==0.1.1
safetensors==0.5.3
sageattention==1.0.6
scikit-learn==1.7.2
scipy==1.15.3
seaborn==0.13.2
segmentation-models-pytorch==0.3.3
sentencepiece==0.2.0
sentry-sdk==2.27.0
setproctitle==1.3.5
shellingham==1.5.4
silero-vad==5.1.2
six==1.17.0
smmap==5.0.2
sniffio==1.3.1
sortedcontainers==2.4.0
soundfile==0.12.1
sox==1.5.0
soxr==1.0.0
spafe==0.3.2
speechbrain==1.0.3
SQLAlchemy==2.0.37
stack-data==0.6.3
submitit==1.5.2
sympy==1.13.1
tabulate==0.9.0
tenacity==9.1.2
tensorboard==2.19.0
tensorboard-data-server==0.7.2
termcolor==3.0.1
text-unidecode==1.3
texterrors==0.5.1
threadpoolctl==3.5.0
timm==0.9.2
tokenizers==0.21.1
tomli==2.2.1
toolz==1.0.0
torch==2.5.1+cu121
torch-audiomentations==0.12.0
torch-log-wmse==0.3.0
torch-optimizer==0.1.0
torch-stoi==0.2.3
torch_pitch_shift==1.2.5
torchaudio==2.5.1+cu121
torchmetrics==0.11.4
torchseg==0.0.1a1
torchvision==0.20.1+cu121
tqdm==4.67.1
traitlets==5.14.3
transformers==4.48.3
treetable==0.2.5
triton-windows==3.4.0.post20
typeguard==4.4.2
typer==0.15.2
typing-inspection==0.4.0
typing_extensions==4.12.2
tzdata==2025.2
umap-learn==0.5.6
unicodedata2==15.1.0
Unidecode==1.4.0
uritemplate==4.1.1
urllib3==2.3.0
uroman==1.3.1.1
wandb==0.19.10
wcwidth==0.2.13
webdataset==0.2.111
websockets==15.0.1
Werkzeug==3.1.3
wespeaker @ git+https://github.com/wenet-e2e/wespeaker.git@f59df4fa3351a6be57493e83d466f42825b8d568
wget==3.2
whisper_normalizer==0.1.12
widgetsnbextension==4.0.14
win32_setctime==1.2.0
wrapt==1.17.2
xxhash==3.5.0
yarl==1.20.0
